src/endpoint/s3/ops/s3_get_bucket.js

```diff
@@ -35,43 +41,54 @@ async function get_bucket(req) {
     }
     const reply = await req.object_sdk.list_objects(params);
 
+    const encoding_type = req.query['encoding-type'];
+    const encoding_flag = set_encoding_flag(encoding_type);
+    const fields_to_encode = {
```

Comment:
I think that `fields_to_encode` can be a misleading name. 
those fields are always returning even when we are not encoding them.
I don't have a great name but maybe `fields_to_return`? 
@shirady WDYT?

Reply:
I would actually avoid extracting fields from the response object. it's just a bit confusing to follow.
I think we can simply prepare an `encode_field` function based on the encoding_type, so that it will refer either to a no_encoding function that simply returns the input, or refer to a url encoding function, something like this:
```js
    const no_encode = v => v;
    const encode_url_field = v => v && encodeURIComponent(v).replaceAll('%20', `+`);
    const encode_field = encoding_type === 'url' ? encode_url_field : no_encode;
    return {
        ListBucketResult: [{
                Name: req.params.bucket,
                Prefix: encode_field(req.query.prefix),
                Delimiter: encode_field(req.query.delimiter || undefined),
                MaxKeys: max_keys_received,
                IsTruncated: reply.is_truncated,
                'Encoding-Type': encoding_type,
...

```

Reply:
Hi,
@liranmauda 'fields_to_encode' is not a perfect name, but it serves the intention that those fields _might_ be encoded - many fields are returning, but those fields are the encoded ones.

@guymguym at first this is what we did and then I changed it to work with a specific object because we thought the calling to functions in the 'return' part is less readable/maintainable (as I commented):

> Note about the implementation: As a premise, we decide to change the encoding **after** we get the reply (as opposed to sending it as a param and changing the implementation of list objects in every namespace).
> 
> There were two options for this implementation:
> 
> 1. Call the function of encoding for every field needed - every time check the flag and if it should be encoded then call the function that encodes the value (notice those fields are inside the return object).
> 2. Create a structure that will have all the fields that should be encoded, it will be encoded as needed (according to the flag).
> 
> second option was chosen for **readability** and **easier maintenance**.

Reply:
i still think the separate object reduces the flow readability so i would prefer an inline encoding calls